### DISCUSSION TITLE
纳秒级高性能日志系统 · ATC 2018 - 面向信仰编程 · /papers-nanolog

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#633] [Announcements]
https://draveness.me/papers-nanolog/

**Comment by dawncold** [comment]
typo: NanoLog 能在“毫秒”尺度打印日志是因为它...

---

*2021-01-31 UPDATES: 已修复*

**Comment by lionxu** [comment]
优化就是权衡和取舍的过程

**Comment by liushiqi9** [comment]
二进制日志是NanoLog的一个特性但是并不是无法避免的， 原作者选择了 二进制日志 是为了性能和DiskSpace， 如果正常使用完全可以在后台Compaction线程中直接进行Decompress 输出Human-readable 的日志。
